docs/develop/python/message-passing.mdx

```diff
@@ -27,155 +27,175 @@ tags:
   - dynamic-handlers
 ---
 
-This page shows how to do the following:
+See [Workflow message passing](/encyclopedia/workflow-message-passing#sending-queries) for an introduction to using messages with Temporal Workflows.
 
-- [Develop with Signals](#signals)
-- [Develop with Queries](#queries)
-- [Develop with Updates](#updates)
-- [Dynamic Handler](#dynamic-handler)
+Here is an example Workflow Definition featuring Query, Signal, and Update handlers, and an Update validator.
+You can view this as a runnable sample at [hello/hello_message_passing.py](https://github.com/temporalio/samples-python/blob/dan/dan/docs_message_passing.py).
 
-## Signals {#signals}
+```python
```

Comment:
These code snippets should be pulled in from external code repositories throughout, not embedded directly.

Reply:
[Agreed via other channels that we'd do the snipsync linkage in subsequent PRs]